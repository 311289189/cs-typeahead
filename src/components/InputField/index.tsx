import React, { InputHTMLAttributes } from 'react';
import './index.css'

interface Props {
    onChange?(value: string): void
    value?: string
}

const InputField = (props: Props & Omit<InputHTMLAttributes<never>, 'onChange'>) => {

    const {
        onChange,
        ...rest
    } = props

    return (
      <input
        onChange={e => onChange && onChange(e.target.value)}
        className="generic-input"
        type="text"
        value={props.value}
        { ...rest }
      />
    )
}

export default InputField
