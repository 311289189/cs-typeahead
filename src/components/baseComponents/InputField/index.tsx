import React, { InputHTMLAttributes } from 'react';

interface Props {
    onChange?(value: string): void
    value?: string
    hasFocus?(hasFocus: boolean): void
}

const InputField = (props: Props & Omit<InputHTMLAttributes<never>, 'onChange'>) => {

    const {
        onChange,
        hasFocus,
        ...rest
    } = props

    return (
      <input
        onChange={e => onChange && onChange(e.target.value)}
        onFocus={() => hasFocus && hasFocus(true)}
        onBlur={() => hasFocus && hasFocus(false)}
        type="text"
        value={props.value}
        { ...rest }
      />
    )
}

export default InputField
