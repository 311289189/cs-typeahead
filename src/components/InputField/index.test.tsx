import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import InputField from './';

const inputTextFixture = 'hello'

test('matches snapshot', () => {
    const { container } = render(<InputField />);
    const inputElement = container.querySelector('input')
    expect(inputElement).toMatchSnapshot();
});

test('renders <Input/> component', () => {
    const { container } = render(<InputField />);
    const inputElement = container.querySelector('input')
    expect(inputElement).toBeInTheDocument();
});

test('<Input/> state can be updated', async () => {
    const label = 'test-label';
    const inputWrapper = render(<InputField aria-label={label} />);
    const input = inputWrapper.getByLabelText(label) as HTMLInputElement;
    fireEvent.change(input, { target: { value: inputTextFixture } });
    expect(input.value).toBe(inputTextFixture);
});

test('when <Input/> internal state is updated callback prop is fired', async () => {
    const spy = jest.fn();
    const label = 'test-label';
    const input = render(<InputField onChange={spy} aria-label={label}/>).getByLabelText(label);
    fireEvent.change(input, { target: { value: inputTextFixture } });
    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy).toHaveBeenCalledWith(inputTextFixture);
});
