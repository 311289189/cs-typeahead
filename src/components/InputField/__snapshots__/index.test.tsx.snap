// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<input
  class="generic-input"
  type="text"
  value=""
/>
`;
